lib/plurimath/math/formula.rb

```diff
@@ -273,6 +275,31 @@ def omml_br_tag
         r_tag = ox_element("r", namespace: "m")
         r_tag << ox_element("br")
       end
+
+      def unitsml_post_processing(nodes)
+        nodes.each.with_index do |node, index|
+          if node.is_a?(Ox::Element) && node.attributes&.dig(:unitsml)
+            previous = nodes[index-1]
+            if previous && ["mi", "mn"].include?(previous.name)
+              if dump_ox_nodes(node.nodes).match?(/>(&#x[0-9a-fA-F]+;)*(\w+|\d+)</)
```

Comment:
This condition isn't right. 

````
text = HTMLEntities.new.encode(Nokogiri::XML("<mrow>#{rendering}</mrow>")
        .text.strip)
      if /\p{L}|\p{N}/.match?(text)
        "<mo rspace='thickmathspace'>&#x2062;</mo>"
      else "<mo>&#x2062;</mo>"
     end
````

This is saying that if, according to unicode, there is a number or a letter in the text (`/\p{L}|\p{N}/`), use a thickmathspace; otherwise, do not. In the case of say 9°, ° is not a letter or a number, therefore there is no space between 9 and °.

But your test is using Ascii-based \w and \d, and matches on any XML escape. That means it will not pick up that Å is a letter (it's not a standard metric unit, but it is still a unit); and if ° is encoded as an escape, `&#x00b0;`, it will not pick up that ° is not a letter or number.

Please do a more sophisticated test. You need to extract the text content of `dump_ox_nodes(node.nodes)` (the equivalent of Nokogiri `nodes.text`), and apply specifically the test `/\p{L}|\p{N}/.match?(HTMLEntities.new.decode(text))` to it, resolving the XML escapes to their Unicode counterparts.

(My source code had `.encode`; but sadly, that is a bug.)

In testing, we expect to see:

* `<mn>9</mn> <mi>Å</mi>` => `<mn>9</mn> <mo space="thickspace">&#x2062;</mo><mi>Å</mi>` : _9 Å_
* `<mn>9</mn> <mi>°</mi>` => `<mn>9</mn> <mo>&#x2062;</mo><mi>°</mi>`: _9°_


You  currently have a test on `'1 "unitsml(um)"'`. Add tests on `'1 "unitsml(Aring)"'` and `'1 "unitsml(deg)"'`. (Symbols documented on https://github.com/plurimath/asciimath2unitsml )

Btw, the symbols documentation appears in Plurimath.org, right?

Reply:
@opoudjis can we merge this now? Thanks.

Reply:
Yes, I reviewed and approved